### PR TITLE
Manifest read size limit

### PR DESCRIPTION
**Pull request by janos**
Resolves https://github.com/ethersphere/go-ethereum/issues/468.

### CODE REVIEW

File: swarm/api/manifest.go [hunk 1]
```diff
@@ -34,6 +34,8 @@ import (
 const (
 	ManifestType        = "application/bzz-manifest+json"
 	ResourceContentType = "application/bzz-resource"
+
+	manifestSizeLimit = 100 * 1024 * 1024
```
Comment (zelig) [inline swarm/api/manifest.go]: i think this is insanely big, since manifests are compacted in a trie, the limit can be much smaller

**Review by zelig — COMMENTED** [review]
adding a test to check the error would be great

**Review by nonsense — APPROVED** [review]

File: swarm/api/manifest.go [hunk 2]
```diff
@@ -220,6 +222,10 @@ func readManifest(manifestReader storage.LazySectionReader, hash storage.Key, dp
 		err = fmt.Errorf("Manifest not Found")
 		return
 	}
+	if size > manifestSizeLimit {
```
Comment (nonsense) [inline swarm/api/manifest.go:225]: Maybe add a error/warning log with the key `hash`?

**Review by zelig — APPROVED** [review]

File: swarm/api/manifest.go (hunk 2 above)
Comment (janos) [reply swarm/api/manifest.go:225]: 👍 Added

**Review by gbalint — APPROVED** [review]

**Pull request merged**